lib/reline/ansi.rb

```diff
@@ -64,18 +65,33 @@ def self.set_default_key_bindings(config, allow_terminfo: true)
     end
   end
 
+  def self.set_default_key_bindings_ansi_cursor(config)
+    ANSI_CURSOR_KEY_BINDINGS.each do |char, (default_func, modifiers)|
+      bindings = [["\e[#{char}", default_func]] # CSI + char
+      if modifiers[:ctrl]
+        # CSI + ctrl_key_modifier + char
+        bindings << ["\e[1;5#{char}", modifiers[:ctrl]]
+      end
+      if modifiers[:meta]
+        # CSI + meta_key_modifier + char
+        bindings << ["\e[1;3#{char}", modifiers[:meta]]
+        # Meta(ESC) + CSI + char
+        bindings << ["\e\e[#{char}", modifiers[:meta]]
+      end
+      bindings.each do |sequence, func|
+        key = sequence.bytes
+        config.add_default_key_binding_by_keymap(:emacs, key, func)
+        config.add_default_key_binding_by_keymap(:vi_insert, key, func)
+        config.add_default_key_binding_by_keymap(:vi_command, key, func)
+      end
+    end
+  end
+
   def self.set_default_key_bindings_terminfo(config)
     key_bindings = CAPNAME_KEY_BINDINGS.map do |capname, key_binding|
       begin
         key_code = Reline::Terminfo.tigetstr(capname)
-        case capname
-        # Escape sequences that omit the move distance and are set to defaults
-        # value 1 may be sometimes sent by pressing the arrow-key.
-        when 'cuu', 'cud', 'cuf', 'cub'
-          [ key_code.sub(/%p1%d/, '').bytes, key_binding ]
```

Comment:
```
# searched cuu from terminfo.src
cuu=\E[%p1%dA
cuu=\233%p1%dA
cuu=\Ep-%p1%d;
cuu=\E&a-%p1%dR
cuu=\E[%p1%dA$<5>
cuu=\E[%p1%dA$<30>
cuu=\037up %p1%d\r
cuu=\035up %p1%d;
```
these are not for input sequence. It's for output sequence like `\e[42A = move cursor up 42 rows`.
This pattern does not match to `\e[A (UP)` nor `\e[1;2A (UP+SHIFT)`.
We need to explicitly bind `"\e[A"`, not `irrelevant_value.gsub(replace_it, to_match_CSI_A)`